src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -141,6 +141,83 @@ describe("provider", () => {
       await provider.disconnect();
     });
 
+    describe("uses timestamp adjustment in subsequent blocks after calling `evm_mine` with a `timestamp` argument", () => {
+      const timestampIncrement = 5; // seconds
+      const timeArgumentSeconds = 100;
+
+      async function mineBlocksForTimestamps(
+        provider: EthereumProvider
+      ): Promise<string[]> {
+        // mine a block with a specified timestamp
+        await provider.request({
+          method: "evm_mine",
+          params: [timeArgumentSeconds]
+        });
+        const specifiedBlock = await provider.request({
+          method: "eth_getBlockByNumber",
+          params: ["latest", false]
+        });
+        // mine a block without a specified timestamp
+        await provider.request({
+          method: "evm_mine",
+          params: []
+        });
+        const unspecifiedBlock = await provider.request({
+          method: "eth_getBlockByNumber",
+          params: ["latest", false]
+        });
+
+        return [specifiedBlock.timestamp, unspecifiedBlock.timestamp];
```

Comment:
If you coerce these to numbers here it'd make the tests slightly simpler.

Reply:
I intentionally didn't do this, thinking that we should be asserting the _exact_ returned value. Otherwise, there's a potential issue that could slip through where the return type isn't what's expected, but _coerces to_ what's expected.

Thinking about it in light of your comment, I think that we _should_ be ensuring the above, but maybe in a specific test for that.

I've updated my tests to what you've suggested, keen to hear your thoughts on that.